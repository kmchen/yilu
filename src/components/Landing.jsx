import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import Avatar from '@material-ui/core/Avatar';
import Typography from '@material-ui/core/Typography';
import { red, green, blue } from '@material-ui/core/colors';
import FlightLandIcon from '@material-ui/icons/FlightLand';
import Paper from '@material-ui/core/Paper';
import Box from '@material-ui/core/Box';
import convertDate from '../util/util';

const useStyles = makeStyles(theme => ({
  root: {
    padding: theme.spacing(1),
    paddingTop: theme.spacing(0),
    paddingBottom: theme.spacing(0),
    borderLeft: '#ddd',
  },
  onTime: {
    color: green[500],
  },
  boardingTime: {
    color: blue[800],
  },
  landingIcon: {
    backgroundColor: red[500],
  },
  box: {
    margin: 'auto',
  },
}));

export default function Landing({
  origin,
  destination,
  gate,
  seat,
  isOnTime,
  boardingTime,
  flightNumber,
}) {
  const classes = useStyles();
  const flightStatus = isOnTime ? 'onTime' : 'Delay';
  const flightStatusStyle = isOnTime ? classes.onTime : '';
  return (
    <Paper className={classes.root} square elevation={0}>
      <Box display="flex" justifyContent="space-between">
        <Typography>{flightNumber}</Typography>
        <Typography className={classes.boardingTime}>{convertDate(boardingTime)}</Typography>
      </Box>
      <Box
        borderTop={1}
        borderBottom={1}
        display="flex"
        flexDirection="row"
        p={1}
        m={1}
        bgcolor="background.paper"
      >
        <Box className={classes.box}>{origin}</Box>
        <Box>
          <Avatar className={classes.landingIcon}>
            <FlightLandIcon />
          </Avatar>
        </Box>
        <Box className={classes.box}>{destination}</Box>
      </Box>
      <Box display="flex" justifyContent="space-between">
        <Typography>{gate}</Typography>
        <Typography>{seat}</Typography>
        <Typography className={flightStatusStyle}>{flightStatus}</Typography>
      </Box>
    </Paper>
  );
}

/* eslint-disable react/forbid-prop-types */
Landing.propTypes = {
  origin: PropTypes.string.isRequired,
  destination: PropTypes.string.isRequired,
  gate: PropTypes.string.isRequired,
  seat: PropTypes.string.isRequired,
  isOnTime: PropTypes.bool.isRequired,
  boardingTime: PropTypes.number.isRequired,
  flightNumber: PropTypes.string.isRequired,
};
