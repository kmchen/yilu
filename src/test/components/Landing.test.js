import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Landing from '../../components/Landing';
import data from '../../../data/data.js';

configure({ adapter: new Adapter() });

describe('Landing component', () => {
  let landingComponent;

  beforeEach(() => {
    const props = {
      origin: 'an origin',
      destination: 'a destination',
      gate: 'a gate',
      seat: 'a seat',
      isOnTime: true,
      boardingTime: 12345667,
      flightNumber: '1234567',
    };
    landingComponent = shallow(<Landing {...props} />);
  });

  it('should render Landing component', () => {
    expect(landingComponent).toMatchSnapshot();
  });
});
