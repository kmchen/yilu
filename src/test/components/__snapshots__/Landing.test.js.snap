// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing component should render Landing component 1`] = `
<WithStyles(ForwardRef(Paper))
  className="makeStyles-root-1"
  elevation={0}
  square={true}
>
  <Styled(MuiBox)
    display="flex"
    justifyContent="space-between"
  >
    <WithStyles(ForwardRef(Typography))>
      1234567
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-boardingTime-3"
    >
      22:21
    </WithStyles(ForwardRef(Typography))>
  </Styled(MuiBox)>
  <Styled(MuiBox)
    bgcolor="background.paper"
    borderBottom={1}
    borderTop={1}
    display="flex"
    flexDirection="row"
    m={1}
    p={1}
  >
    <Styled(MuiBox)
      className="makeStyles-box-5"
    >
      an origin
    </Styled(MuiBox)>
    <Styled(MuiBox)>
      <WithStyles(ForwardRef(Avatar))
        className="makeStyles-landingIcon-4"
      >
        <FlightLandIcon />
      </WithStyles(ForwardRef(Avatar))>
    </Styled(MuiBox)>
    <Styled(MuiBox)
      className="makeStyles-box-5"
    >
      a destination
    </Styled(MuiBox)>
  </Styled(MuiBox)>
  <Styled(MuiBox)
    display="flex"
    justifyContent="space-between"
  >
    <WithStyles(ForwardRef(Typography))>
      a gate
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))>
      a seat
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-onTime-2"
    >
      onTime
    </WithStyles(ForwardRef(Typography))>
  </Styled(MuiBox)>
</WithStyles(ForwardRef(Paper))>
`;
